extension/src/popup/views/ViewPublicKey/index.tsx

```diff
@@ -0,0 +1,141 @@
+
+import { Form } from "popup/basics/Forms";
+import { ROUTES } from "popup/constants/routes";
+import { METRIC_NAMES } from "popup/constants/metricsNames";
+import { navigateTo, openTab } from "popup/helpers/navigate";
+import { BackButton } from "popup/basics/Buttons";
+import { PopupWrapper } from "popup/basics/PopupWrapper";
+import { BottomNav } from "popup/components/BottomNav";
+import {
+  accountNameSelector,
+  publicKeySelector,
+  updateAccountName,
+} from "popup/ducks/accountServices";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+
+import { Button, TextLink, Icon, Input } from "@stellar/design-system";
+
+import "./styles.scss";
+
+export const ViewPublicKey = () => {
+  const publicKey = useSelector(publicKeySelector);
+  const accountName = useSelector(accountNameSelector);
+  const [isCopied, setIsCopied] = useState(false);
+  const [isEditingName, setIsEditingName] = useState(false);
+  const { network } = useSelector(settingsNetworkDetailsSelector);
+
+  const dispatch = useDispatch();
+
+  interface FormValue {
+    accountName: string;
+  }
+
+  const initialValues: FormValue = {
+    accountName,
+  };
+
+  const handleSubmit = async (values: FormValue) => {
+    const { accountName: newAccountName } = values;
+    if (accountName !== newAccountName) {
+      await dispatch(updateAccountName(newAccountName));
+      emitMetric(METRIC_NAMES.viewPublicKeyAccountRenamed);
+    }
+    setIsEditingName(false);
+  };
+
+  const handleCopy = () => {
+    setIsCopied(true);
+    emitMetric(METRIC_NAMES.viewPublicKeyCopy);
+
+    const t = setTimeout(() => {
+      setIsCopied(false);
+      clearTimeout(t);
+    }, 1000);
+  };
+
+  return (
+    <PopupWrapper>
+      <div className="ViewPublicKey">
+        <BackButton isPopup onClick={() => navigateTo(ROUTES.account)} />
+        <div className="ViewPublicKey__header">
+          {isEditingName ? (
+            <Formik initialValues={initialValues} onSubmit={handleSubmit}>
+              <Form className="ViewPublicKey__form">
+                <Field name="accountName">
+                  {({ field }: FieldProps) => (
+                    <Input
+                      autoComplete="off"
+                      id="accountName"
+                      placeholder={accountName}
+                      {...field}
+                    />
+                  )}
+                </Field>
+                <button
+                  className="ViewPublicKey--account-name-btn"
+                  type="submit"
+                >
+                  <Icon.Check />
+                </button>
+              </Form>
+            </Formik>
+          ) : (
+            <>
+              <div>{accountName}</div>
+              <button
+                className="ViewPublicKey--account-name-btn"
+                onClick={() => setIsEditingName(true)}
+              >
+                <Icon.Edit />
+              </button>
+            </>
+          )}
+        </div>
```

Comment:
Can you make the `Back` arrow lineup with the Account name and `Edit/Check` button so they're all in a line?

Reply:
Also, pretty NIT: but I noticed when the Account Name isn't being edited, it's center aligned, but when you click Edit, the Account Name is left aligned in the input box. Can you make it centered in the input box, please? I think it'll make the experience a little less jumpy when moving between Edit and Save

Reply:
>Can you make the Back arrow lineup with the Account name and Edit/Save button so they're all in a line?

I was gonna wait for the[ better BackButton](https://github.com/stellar/freighter/pull/372#discussion_r815024523) you're adding before making sure it looked exactly right. What about a TODO for when that component's added? (or has it already been added?)

>Can you make it centered in the input box, please? I think it'll make the experience a little less jumpy when moving between Edit and Save

yup I agree. I also changed the display height to match the input which I think makes less jumpy as well. Take a look if you get a sec
https://github.com/stellar/freighter/pull/375/commits/7c2650980c6c57954caa70031921affd8a93513e

Reply:
caused some styling errors fixed here: https://github.com/stellar/freighter/pull/375/commits/58945962b1510042c937b5bca05a73fc89b0494e

Reply:
re: better backButton

Good call - we can wait on this. Let's add a TODO so we don't forget :)